salt/edx/mitx_etl.sls

```diff
@@ -17,28 +16,29 @@ clone_mitx_etl_repo:
     - target: /mitx
     - force_clone: True
     - require:
-      - pkg: install_python_3
+      - pkg: install_etl_os_depencies
+      - file: create_mitx_directory
 
 install_mitx_residential_etl_requirements:
   virtualenv.managed:
     - name: /mitx/mitx_etl
     - system_site_packages: False
     - requirements: /mitx/requirements.txt
+    - python: /usr/bin/python3
     - env_vars:
         PATH_VAR: '/usr/local/bin/pip3'
     - require:
       - git: clone_mitx_etl_repo
-      - pkg: install_python3_pip
+      - pkg: install_etl_os_depencies
 
-{% for key, value in salt.pillar.get('edx:mitx_etl:mitx_residential_etl:settings', {}).items() %}
+{% set settings = salt.pillar.get('edx:mitx_etl:mitx_residential_etl:settings', {}).items() %}
```

Comment:
You don't want to have the `.items()` call at the end, it should just be the `pillar.get` call.